### ISSUE TITLE
journalctl: I/O error when a journal file is empty

### ISSUE THREAD

**Issue by philips** [coreos/bugs#1244]
Empty journal files cause un-debuggable errors. This can happen in an out-of-disk scenario and has caused issues for users in the past. 

There are two things we need to fix:

1) Make journalctl resilient to these empty files and tell the user what to do instead of simply returning I/O error

2) Figure out how to make sure the journald doesn't create these types of files in the future. We don't have a solid repro case right now.

```
localhost core # cat > /var/log/journal/
d14db848889a4d6d8fe6d4be3af67dd3/ remote/
localhost core # cat > /var/log/journal/d14db848889a4d6d8fe6d4be3af67dd3/foo.journal
localhost core # journalctl -f
Error was encountered while opening journal files: Input/output error
localhost core # rm /var/log/journal/d14db848889a4d6d8fe6d4be3af67dd3/foo.journal
localhost core # journalctl -f
-- Logs begin at Tue 2016-04-26 17:46:48 UTC. --
```

Host information:

```
localhost core # cat /etc/os-release
VERSION_ID=1010.1.0
localhost core # systemctl --version
systemd 225
-PAM +AUDIT +SELINUX +IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT -GNUTLS -ACL +XZ -LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD -IDN
```

**Comment by philips** [comment]
With systemd 229 the error will be:

```
Journal file /var/log/journal/12fb54ed01694ec2a58c3be44dc228f9/foo.journal is truncated, ignoring file.
```

Should be fixed in CoreOS Alpha today:1029.0.0 or greater

**Comment by mischief** [comment]
FTR the pr that appears to fix this is:

- https://github.com/systemd/systemd/pull/1758
- systemd/systemd@4f52b822b05c373f40fea1a41ae3ade5d5ff558e
